tests/lib/nm/dns_test.py

```diff
@@ -135,72 +142,123 @@ def _assert_dns(setting_ip, dns_conf):
     assert setting_ip.props.dns_priority == priority
 
 
-def test_find_interfaces_for_dns_with_ipv6_and_ipv4_static_gateways():
-    iface_routes = _get_test_iface_routes()
-    assert (
-        TEST_IPV4_GATEWAY_IFACE,
-        TEST_IPV6_GATEWAY_IFACE,
-    ) == nm_dns.find_interfaces_for_name_servers(iface_routes)
-
-
-def test_find_interfaces_for_dns_with_ipv6_static_gateway_only():
-    iface_routes = {
-        TEST_IPV6_GATEWAY_IFACE: _get_test_ipv6_gateway(),
-        TEST_STATIC_ROUTE_IFACE: _get_test_static_routes(),
-    }
-    assert (
-        None,
-        TEST_IPV6_GATEWAY_IFACE,
-    ) == nm_dns.find_interfaces_for_name_servers(iface_routes)
+parametrize_ip_ver = pytest.mark.parametrize(
+    "families",
+    [(Interface.IPV4,), (Interface.IPV6,), (Interface.IPV4, InterfaceIPv6)],
+    ids=["ipv4", "ipv6", "ipv4&ipv6"],
+)
 
 
-def test_find_interfaces_for_dns_with_ipv4_static_gateway_only():
-    iface_routes = {
-        TEST_IPV4_GATEWAY_IFACE: _get_test_ipv4_gateway(),
-        TEST_STATIC_ROUTE_IFACE: _get_test_static_routes(),
-    }
-    assert (
-        TEST_IPV4_GATEWAY_IFACE,
-        None,
-    ) == nm_dns.find_interfaces_for_name_servers(iface_routes)
+@parametrize_ip_ver
+def test_find_interfaces_for_dns_with_static_gateways(families):
+    state = _get_test_desired_state_static_gateway(families)
+    expected_ifaces = [None, None]
+    if Interface.IPV4 in families:
+        expected_ifaces[0] = TEST_IPV4_GATEWAY_IFACE
+    if Interface.IPV6 in families:
+        expected_ifaces[1] = TEST_IPV6_GATEWAY_IFACE
+    expected_ifaces = tuple(expected_ifaces)
+    assert expected_ifaces == nm_dns.find_interfaces_for_name_servers(
+        State(state)
+    )
 
 
 def test_find_interfaces_for_dns_with_no_routes():
-    iface_routes = {}
-    assert (None, None) == nm_dns.find_interfaces_for_name_servers(
-        iface_routes
+    state = State(
+        {
+            Interface.KEY: [
+                {
+                    Interface.NAME: TEST_IPV4_GATEWAY_IFACE,
+                    Interface.STATE: InterfaceState.UP,
+                },
+                {
+                    Interface.NAME: TEST_IPV6_GATEWAY_IFACE,
+                    Interface.STATE: InterfaceState.UP,
+                },
+            ],
+        }
     )
+    assert (None, None) == nm_dns.find_interfaces_for_name_servers(state)
 
 
 def test_find_interfaces_for_dns_with_no_gateway():
-    iface_routes = {TEST_STATIC_ROUTE_IFACE: _get_test_static_routes()}
+    state = _get_test_desired_state_static_gateway(())
```

Comment:
nit: I think it would be better to do `State(_get_test_desired_state_static_gateway())`